datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/mutate/util/GlossaryUtils.java

```diff
@@ -0,0 +1,102 @@
+import com.linkedin.datahub.graphql.authorization.DisjunctivePrivilegeGroup;
+import com.linkedin.entity.EntityResponse;
+import com.linkedin.entity.client.EntityClient;
+import com.linkedin.glossary.GlossaryNodeInfo;
+import com.linkedin.glossary.GlossaryTermInfo;
+import com.linkedin.metadata.Constants;
+import com.linkedin.metadata.authorization.PoliciesConfig;
+import com.linkedin.r2.RemoteInvocationException;
+import lombok.extern.slf4j.Slf4j;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.net.URISyntaxException;
+import java.util.Optional;
+
+import static com.linkedin.datahub.graphql.resolvers.AuthUtils.ALL_PRIVILEGES_GROUP;
+
+@Slf4j
+public class GlossaryUtils {
+
+  private GlossaryUtils() { }
+
+  public static boolean canManageGlossaries(@Nonnull QueryContext context) {
+    return AuthorizationUtils.isAuthorized(context, Optional.empty(), PoliciesConfig.MANAGE_GLOSSARIES_PRIVILEGE);
+  }
+
+  /**
+   * Returns true if the current user is able to create, delete, or move Glossary Terms and Nodes under a parent Node.
+   * They can do this with either the global MANAGE_GLOSSARIES privilege, or if they have the MANAGE_CHILDREN privilege
+   * on the relevant parent node in the Glossary.
+   */
+  public static boolean canManageGlossaryEntity(@Nonnull QueryContext context, @Nullable Urn parentNodeUrn) {
+    if (canManageGlossaries(context)) {
+      return true;
+    }
+    if (parentNodeUrn == null) {
+      return false; // if no parent node, we must rely on the canManageGlossaries method above
+    }
+
+    final DisjunctivePrivilegeGroup orPrivilegeGroups = new DisjunctivePrivilegeGroup(ImmutableList.of(
+        ALL_PRIVILEGES_GROUP,
+        new ConjunctivePrivilegeGroup(ImmutableList.of(PoliciesConfig.MANAGE_CHILDREN_PRIVILEGE.getType()))
+    ));
+
+    return AuthorizationUtils.isAuthorized(
+        context.getAuthorizer(),
+        context.getActorUrn(),
+        parentNodeUrn.getEntityType(),
+        parentNodeUrn.toString(),
+        orPrivilegeGroups);
+  }
+
+  public static Urn getTermParentUrn(Urn termUrn, QueryContext context, EntityClient entityClient) {
+    try {
+      EntityResponse response = entityClient.getV2(Constants.GLOSSARY_TERM_ENTITY_NAME, termUrn,
+          ImmutableSet.of(Constants.GLOSSARY_TERM_INFO_ASPECT_NAME), context.getAuthentication());
+      if (response != null && response.getAspects().get(Constants.GLOSSARY_TERM_INFO_ASPECT_NAME) != null) {
+        GlossaryTermInfo termInfo = new GlossaryTermInfo(response.getAspects()
+            .get(Constants.GLOSSARY_TERM_INFO_ASPECT_NAME).getValue().data());
+        return termInfo.getParentNode();
+      }
+      return null;
+    } catch (URISyntaxException | RemoteInvocationException e) {
+      throw new RuntimeException("Failed to fetch Glossary Term to check for privileges", e);
+    }
+  }
+
+  public static Urn getNodeParentUrn(Urn nodeUrn, QueryContext context, EntityClient entityClient) {
```

Comment:
minor: java doc.

We can also annotate this method as @Nullable. 

Also prefer to annotate input args where possible